shell/key-bindings.fish

```diff
@@ -84,4 +86,26 @@ function fzf_key_bindings
     bind -M insert \cr fzf-history-widget
     bind -M insert \ec fzf-cd-widget
   end
+
+  function __fzf_get_starting_dir
+    set -l dir (commandline -t)
+
+    # Default to current dir if no argument input
+    if [ -z "$dir" ]
+        set dir '.'
+    end
+
+    # The commandline token might be escaped, we need to unescape it.
+    set dir (eval "printf '%s' $dir")
+
+    if [ ! -d "$dir" ]
+      return 1
```

Comment:
> This is a helper function and, imo, it should defer responsibility to the file/cd widget to respond to invalid dir.

But the error is not properly handled outside the function and that's why I pointed it out.

> which results in a user-intuitive behaviour of no fzf completions

Actually I'm a bit torn on this. Because currently `alt-c` ignores whatever is on the command line, so changing it to produce an empty list somehow breaks backward compatibility. (e.g. a user types in `git add`, then realizes that she wants to change the directory so immediately presses `alt-c`, without having to delete the command line.) `alt-c` of bash and zsh also have the same behavior. 

> I think this might result in invalid filepaths for file widget.

Similarly, that's how file widget currently behaves.